h2o-algos/src/test/java/hex/deeplearning/DeepLearningAutoEncoderTest.java

```diff
@@ -151,12 +157,37 @@ public void run() {
           Assert.assertTrue(outliers.contains(new Long(21)));
           Assert.assertTrue(outliers.contains(new Long(22)));
           Assert.assertTrue(outliers.size() == 3);
+
+          // check if reconstruction error is the same from model and mojo model too. Testcase for PUBDEV-6030.
+          try {
+            DeeplearningMojoModel mojoModel = (DeeplearningMojoModel) mymodel.toMojo();
+            EasyPredictModelWrapper model = new EasyPredictModelWrapper(mojoModel);
+            AutoEncoderModelPrediction tmpPrediction;
+            RowData tmpRow;
+            double calcNormMse = 0;
+            for (int r = 0; r < train.numRows(); r++) {
+              tmpRow = new RowData();
+              for (int c = 0; c < train.numCols(); c++) {
+                tmpRow.put(train.names()[c], train.vec(c).at(r));
+              }
+              tmpPrediction = model.predictAutoEncoder(tmpRow);
+              calcNormMse += tmpPrediction.mse;
+            }
+            double mojoMeanError = calcNormMse/train.numRows();
+            sb.append("Mojo mean reconstruction error (train): ").append(mojoMeanError).append("\n");
+            Assert.assertTrue("Mean reconstruction error should be the same from model compate to mojo model " +
```

Comment:
Maybe `assertEquals` with a tolerance would be better?

Reply:
Yes, definitely.